sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -804,8 +806,78 @@ private static bool SupportsDeferredBinding(CustomAttribute attribute)
+                    // InputConverterAttribute will have supported converter type
+                    foreach (var customAttribute in bindingAttribute.ConstructorArguments)
+                    {
+                        if (DoesConverterSupportDeferredBinding(customAttribute, bindingType))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool DoesConverterSupportDeferredBinding(CustomAttributeArgument customAttribute, TypeReference bindingType)
+        {
+            var typeReferenceValue = customAttribute.Value as TypeReference;
+            var typeReferenceCustomAttributes = typeReferenceValue?.Resolve().CustomAttributes;
+            
+            // Attributes advertised by converter
+            if (typeReferenceCustomAttributes is not null)
+            {
+                bool converterAdvertisesDeferredBindingSupport = typeReferenceCustomAttributes.Any(a => string.Equals(a.AttributeType.FullName, Constants.SupportsDeferredBindingAttributeType, StringComparison.Ordinal));
+
+                if (converterAdvertisesDeferredBindingSupport)
+                {
+                    bool converterAdvertisesTypes = typeReferenceCustomAttributes.Any(a => string.Equals(a.AttributeType.FullName, Constants.SupportedConverterTypeAttributeType, StringComparison.Ordinal));
+
+                    if (!converterAdvertisesTypes)
+                    {
+                        // If a converter advertises deferred binding but does not explictly advertise any types then DeferredBinding will be supported for all the types
+                        return true;
+                    }
+                    else if (DoesConverterSupportTargetType(typeReferenceCustomAttributes, bindingType))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool DoesConverterSupportTargetType(Collection<CustomAttribute> customAttributes, TypeReference bindingType)
+        {
+            // Parse attributes advertised by converter
+            foreach (CustomAttribute attribute in customAttributes)
+            {
+                if (string.Equals(attribute.AttributeType.FullName, Constants.SupportedConverterTypeAttributeType, StringComparison.Ordinal))
+                {
+                    foreach (CustomAttributeArgument element in attribute.ConstructorArguments)
+                    {
+                        if (element.Type.FullName == typeof(Type).FullName)
+                        {
+                            var supportedType = element.Value as TypeReference;
+
+                            if (supportedType is not null
+                                && supportedType != null && string.Equals(supportedType.FullName, bindingType.FullName, StringComparison.Ordinal))
```

Comment:
`supportedType is not null && supportedType != null `

Aren't these doing the exact same check?

Reply:
I can't find what `StringComparison` pattern matching uses, but this can also be written as:

``` CSharp
if (supportedType is { FullName: bindingType.FullName })
```